browser-extension/src/views/AddLead.js

```diff
@@ -0,0 +1,286 @@
+import CSSModules from 'react-css-modules';
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+    DateInput,
+    SelectInput,
+    MultiSelectInput,
+    TextInput,
+} from '../public-components/Input';
+
+import { FgRestBuilder } from '../public/utils/rest';
+
+import { LoadingAnimation } from '../public-components/View';
+import {
+    TransparentButton,
+    PrimaryButton,
+} from '../public-components/Action';
+
+import {
+    updateInputValueAction,
+    clearInputValueAction,
+    setProjectListAction,
+    setCurrentTabInfoAction,
+    inputValuesForTabSelector,
+    currentTabIdSelector,
+    projectListSelector,
+    setLeadOptionsAction,
+    leadOptionsSelector,
+} from '../common/redux';
+
+import {
+    createUrlForProjectList,
+    createParamsForProjectList,
+
+    createUrlForLeadOptions,
+    createParamsForLeadOptions,
+
+    urlForLeadCreate,
+    createParamsForLeadCreate,
+} from '../common/rest';
+
+import styles from '../stylesheets/add-lead.scss';
+
+const mapStateToProps = state => ({
+    inputValues: inputValuesForTabSelector(state),
+    currentTabId: currentTabIdSelector(state),
+    projects: projectListSelector(state),
+    leadOptions: leadOptionsSelector(state),
+});
+
+const mapDispatchToProps = dispatch => ({
+    updateInputValue: params => dispatch(updateInputValueAction(params)),
+    clearInputValue: params => dispatch(clearInputValueAction(params)),
+    setCurrentTabInfo: params => dispatch(setCurrentTabInfoAction(params)),
+    setProjectList: params => dispatch(setProjectListAction(params)),
+    setLeadOptions: params => dispatch(setLeadOptionsAction(params)),
+});
+
+@connect(mapStateToProps, mapDispatchToProps)
+@CSSModules(styles, { allowMultiple: true })
+export default class AddLead extends React.PureComponent {
+    constructor(prop) {
+        super(prop);
+
+        this.state = {
+            pending: false,
```

Comment:
Pending should always be "true" as start. (else it will cause flashing problem, clearly evident when restrequest delay is high)